### ISSUE TITLE
Alkalinity values with Extended UNIQUAC

### ISSUE THREAD

**Issue by volpatto** [reaktoro/reaktoro#372]
Hi, Allan.

Just reporting that I have been facing some difficulties to calculate alkalinity values using the Extended UNIQUAC. The other models I have tested worked fine (HKF and Pitzer, basically), showing the expected result from [the paper](https://epic.awi.de/id/eprint/16138/1/Wol2007a.pdf) that we got the alkalinity expression.

Here is a minimal working example demonstrating the issue:

[Alkalinity-calculation-issue-with-EUNIQUAC.zip](https://github.com/reaktoro/reaktoro/files/14517140/Alkalinity-calculation-issue-with-EUNIQUAC.zip)

This case is described in the paper, see the section 2. The titration is not included, only the initial carbonate system.

I am not sure if I would be able to work on the solution of this issue soon, maybe next month I could lend you a hand, but please feel free to address it, I would be very happy to have a solution anyway.

**Comment by allanleal** [comment]
Hi Diego, the issue is because the extended UNIQUAC model does not contain standard molar volumes for the aqueous species and it produces zero excess molar volume for the aqueous solution. As a result, volume of the phase is zero, and the alkalinity implementation depends on a non-zero volume. A temporary solution could be producing an aqueous solution volume that is equal to that of pure water. Ideally, the extended UNIQUAC model should be fundamentally revised to consider these volumetric properties.

**Comment by volpatto** [comment]
> Hi Diego, the issue is because the extended UNIQUAC model does not contain standard molar volumes for the aqueous species and it produces zero excess molar volume for the aqueous solution. As a result, volume of the phase is zero, and the alkalinity implementation depends on a non-zero volume. A temporary solution could be producing an aqueous solution volume that is equal to that of pure water. Ideally, the extended UNIQUAC model should be fundamentally revised to consider these volumetric properties.

Hi Allan,

Ahhhh, yes, indeed, we have to get the Volume from the phase (using an `AqueousProps`), of course. My bad, I didn't pay enough attention. I have been thinking on using the pure water phase volume as a placeholder as well since the solutes are just a really small fraction of the phase. Also considering to implement the IAPWS-08 (for seawater) in Reaktoro (there is a C++ code available). What are your thoughts? The IAPWS is made to be used altogether, using it just for volumes and density may introduce an inconsistency, but should work fine as a placeholder.

**Comment by allanleal** [comment]
The EOS for seawater would give more sensible densities compared to that of pure water. However, it would not account for other dissolved species that also affect the solution density (e.g., CO2, HCO3-, CO3-2).

**Comment by volpatto** [comment]
> The EOS for seawater would give more sensible densities compared to that of pure water. However, it would not account for other dissolved species that also affect the solution density (e.g., CO2, HCO3-, CO3-2).

Yes, indeed. I am not sure if the IAPWS 08 considers carbonates properly, at least for geochemistry applications. Since this is a model for oceanography, they probably address carbonates (due to ocean acidification) up to a limit, I'am not sure. Would you be happy with a temp solution using the Wagner & Pruss model to retrieve the approximation of the aqueous solution volume?
